packages/react-components/react-timepicker-compat-preview/src/components/TimePicker/useTimePicker.tsx

```diff
@@ -122,3 +156,66 @@ const useStableDateAnchor = (providedDate: Date | undefined, startHour: Hour, en
 
   return { dateStartAnchor, dateEndAnchor };
 };
+
+/**
+ * Mimics the behavior of the browser's change event for a freeform TimePicker.
+ * The provided callback is called when input changed and:
+ * - Enter/Tab key is pressed on the input.
+ * - TODO: TimePicker loses focus, signifying a possible change.
+ */
+const useSelectTimeFromValue = (state: TimePickerState, callback: TimePickerProps['onTimeSelect']) => {
+  const {
+    activeOption,
+    dateToText,
+    freeform,
+    validateFreeFormTime,
+    options,
+    selectedTimeTextRef,
+    setActiveOption,
+    setValue,
+    value,
+  } = state;
+
+  // Base Combobox has activeOption default to first option in dropdown even if it doesn't match input value, and Enter key will select it.
+  // This effect ensures that the activeOption is cleared when the input doesn't match any option.
+  // This behavior is specific to a freeform TimePicker where the input value is treated as a valid time even if it's not in the dropdown.
+  const isValueOptionPrefix = value ? options.some(({ text }) => text.indexOf(value) === 0) : false;
+  React.useEffect(() => {
+    if (freeform && value && !isValueOptionPrefix) {
+      setActiveOption(undefined);
+    }
+  }, [freeform, isValueOptionPrefix, setActiveOption, value]);
+
+  const selectTimeFromValue = React.useCallback(
+    (e: React.KeyboardEvent<HTMLInputElement> | React.FocusEvent<HTMLInputElement>) => {
+      if (!freeform) {
+        return;
+      }
+
+      const { date: selectedTime, error } = validateFreeFormTime(value);
+      const selectedTimeText = selectedTime ? dateToText(selectedTime) : value;
+
+      if (selectedTimeText !== value) {
```

Comment:
I tried to get to a case where `selectedTimeText` and `value` are different but I failed. Could you give me an example where this happens?

Reply:
btw nice catch!